From: Stefano Zacchiroli <zack@debian.org>
Date: Tue, 6 Oct 2009 15:34:00 +0200
Subject: Add rpath to Apache module

Add rpath to Apache module, pointing to `ocamlc -where`. Rationale:
ensure libcamlrun_shared.so can be found.

Signed-off-by: Stefano Zacchiroli <zack@debian.org>
---
 src/netcgi2-apache/Makefile.def |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/netcgi2-apache/Makefile.def b/src/netcgi2-apache/Makefile.def
index 7b1bb61..e93bae8 100644
--- a/src/netcgi2-apache/Makefile.def
+++ b/src/netcgi2-apache/Makefile.def
@@ -50,7 +50,7 @@ endif
 ### Only works with OCaml >= 3.11 on plafforms where PIC differs from non-PIC
 ### See http://caml.inria.fr/mantis/view.php?id=3866
 mod_netcgi_apache.so: $(MOD_OBJECTS)
-	$(APXS) -c -o $@ $^ -L$(APACHE_OCAMLLIBDIR) $(patsubst -lcamlrun,-lcamlrun_shared,$(APACHE_OCAMLLIBS))
+	$(APXS) -c -o $@ $^ -L$(APACHE_OCAMLLIBDIR) -Wl,--rpath,$(APACHE_OCAMLLIBDIR) $(patsubst -lcamlrun,-lcamlrun_shared,$(APACHE_OCAMLLIBS))
 	test -f .libs/$@ && cp .libs/$@ .
 
 netcgi_apache_mod.lo: netcgi_apache_mod.o
-- 
